src/coordinator/executor/transform/controller.go

```diff
@@ -51,12 +50,6 @@ func (t *Controller) Process(block storage.Block) error {
 
 // BlockBuilder returns a BlockBuilder instance with associated metadata
 // nolint: unparam
-func (t *Controller) BlockBuilder(blockMeta storage.BlockMetadata) (BlockBuilder, error) {
-	return nil, errors.ErrNotImplemented
-}
-
-// BlockBuilder builds a new block
-type BlockBuilder interface {
-	AppendValue(index int, value float64)
-	Build() storage.Block
+func (t *Controller) BlockBuilder(blockMeta block.Metadata) (block.Builder, error) {
```

Comment:
A bit odd that this is called on a `Controller` if it doesn't do anything with it

Reply:
lets you switch out different implementations, but yeah its a bit early stage, might move it around later.